app/src/main/java/ch/epfl/polybazaar/utilities/ImageTaker.java

```diff
@@ -0,0 +1,170 @@
+package ch.epfl.polybazaar.utilities;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.content.SharedPreferences;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.net.Uri;
+import android.os.Environment;
+import android.provider.MediaStore;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.core.content.FileProvider;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import ch.epfl.polybazaar.R;
+import ch.epfl.polybazaar.widgets.permissions.PermissionRequest;
+
+import static ch.epfl.polybazaar.utilities.ImageUtilities.convertBitmapToStringWithQuality;
+
+public class ImageTaker extends AppCompatActivity {
+
+    public static final int QUALITY = 25;
+    public static final String STRING_IMAGE = "bitmap_image";
+    public static final String PICTURE_PREFS = "bitmap_prefs";
+    public static final String CODE = "request_code";
+    public static final int LOAD_IMAGE = 3;
+    public static final int TAKE_IMAGE = 4;
+    private static final int RESULT_LOAD_IMAGE = 1;
+    private static final int RESULT_TAKE_PICTURE = 2;
+    public static final String BITMAP_OK = "bitmap_image_present";
+    private File photoFile;
+    private Bitmap image;
+    private PermissionRequest cameraPermissionRequest;
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if(resultCode == Activity.RESULT_OK) {
+            if (requestCode == RESULT_LOAD_IMAGE) {
+                if (data == null) {
+                    failure();
+                    return;
+                }
+                Uri selectedImage = data.getData();
+                Bitmap bitmap;
+                try {
+                    bitmap = MediaStore.Images.Media.getBitmap(this.getContentResolver(), selectedImage);
+                } catch (IOException e) {
+                    e.printStackTrace();
+                    Toast.makeText(this, R.string.unable_load_image, Toast.LENGTH_SHORT).show();
+                    failure();
+                    return;
+                }
+                image = bitmap;
+                success();
+            } else if (requestCode == RESULT_TAKE_PICTURE) {
+                ByteArrayOutputStream bin = new ByteArrayOutputStream();
+                image = BitmapFactory.decodeFile(getPhotoFile().getAbsolutePath());
+                image.compress(Bitmap.CompressFormat.JPEG, QUALITY, bin);

```

Comment:
Not PNG here?

Reply:
No, turns out we do need some compression, or else taking a picture and saving it takes around 10 seconds.